### ISSUE TITLE
'vue/comment-directive' — error in the end of public/index.html just after vue-cli installation

### ISSUE THREAD

**Issue by 4erkashin** [vuejs/eslint-plugin-vue#1355]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: 6.8.0** 
- **eslint-plugin-vue version: 7.0.0-0** 
- **Node version: 12.14.1** 
- **Operating System: Ubuntu 20.04.1 LTS** 

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    node: true
  },
  extends: ['plugin:vue/vue3-essential', 'eslint:recommended', '@vue/prettier'],
  parser: 'vue-eslint-parser',
  parserOptions: {
    parser: 'babel-eslint'
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off'
  }
};
```

**What did you do?**	
Just created project with `vue create`

**What did you expect to happen?**
No linting errors

**What actually happened?**

```
error: clear (vue/comment-directive) at public/index.html:21:8:
  19 |     <!-- built files will be auto injected -->
  20 |   </body>
> 21 | </html>
     |        ^
  22 |
```

**Comment by vanderb** [comment]
Same here.
Only solution for me was to disable the rule

````
  rules: {
    'vue/comment-directive': 'off'
  }
````

**Comment by 4erkashin** [comment]
@vanderb yeah. Did the same too. But it looks like a kludge.

**Comment by kawazoe** [comment]
You should at least use an override specific to html files for now since you probably still want that rule in vue files:

```
    {
      // HACK: https://github.com/vuejs/eslint-plugin-vue/issues/1355
      files: ['**/*.html'],
      rules: { 'vue/comment-directive': 'off' },
    },
```

**Comment by ota-meshi** [comment]
Why do you lint HTML with ESLint? eslint-plugin-vue supports linting `.vue`, but not` .html`.

If you really want to lint the HTML, you need to apply the processor to `.html`.

https://eslint.org/docs/developer-guide/working-with-plugins#specifying-processor-in-config-files

I haven't actually run it and confirmed it, but I think it will work if you set it as follows.

```json5
  "overrides": [
    {
      "files": ["*.html"],
      "processor": "vue/.vue"
    }
  ]
```

**Comment by 4erkashin** [comment]
@ota-meshi I really don't know should I use it for `.html` or not. Unfortunately, I'm not so well experienced yet.

But I scaffolded app from CLI and wish there weren't errors. So, could you please explain me, how to configure ESLint correctly? Is it enough to use override from your answer or there is something else I should know?

**Comment by kawazoe** [comment]
@ota-meshi The problem seems to be caused by the rule being applied to plain html files. If you look at the OP, you'll see that it mentions the public/index.html. This rule only make sense in the context of a vue file anyway, so turning it off for html files, for which it shouldn't have been enabled in the first place, make sense.

I am not saying we should enable it for html files. I am saying it IS enabled out of the box for them and shouldn't.

Adding the override I mentioned as is fixes the issue while keeping the rule enabled for vue files.

**Comment by ota-meshi** [comment]
ESLint does not lint `.html` by default. Why do you lint `.html` with ESLint?
If you don't need to apply ESLint to `.html`, I think you should exclude `.html`. I think you're probably targeting `.html` somewhere in the settings.

I don't understand English, so I may not understand what you are saying correctly.

**Comment by kawazoe** [comment]
I think I know what is going on. 

@4erkashin Are you using WebStorm?

I noticed that when creating a new project with `npx @vue/cli create foobar` and then running `npm run lint`, everything is good.

![Windows PowerShell 11_30_2020 7_50_20 PM](https://user-images.githubusercontent.com/406575/100682953-49de5680-3345-11eb-9069-8b2d3492881a.png)

If I open the `public/index.html` file in WebStorm, I get the ESLint error in the IDE. I think it might be WebStorm trying to lint every file you open.

![foobar – index html 11_30_2020 7_49_54 PM](https://user-images.githubusercontent.com/406575/100682928-3b903a80-3345-11eb-9fd4-1f68d67738f5.png)

I still don't know why `vue/comment-directive` is the only rule that triggers in this case.

**Comment by 4erkashin** [comment]
@kawazoe yeah I'm using webstorm. Got it, thanks for explanation)

Should I close the issue? Looks like jetbrains related only.

**Comment by kawazoe** [comment]
It's probably a JetBrains issue.

My workaround will still work for you if you want to use it while we wait for a fix from them.

**Comment by 4erkashin** [comment]
@kawazoe https://youtrack.jetbrains.com/issue/WEB-48468

And again) Explain me please, should I close issue here or not?)

**Comment by piotrtomiak** [comment]
@4erkashin it indeed looks like WebStorm issue. We are going to fix it on our side, you can close the issue.

**Comment by Tricky-Ricky** [comment]
I have this issue on a fresh vue cli install, Using Visual studio code... So I don't think it is explicitly to do with webstorm.
I have also added the rule:
`"vue/comment-directive": 0` 

And the issue still persists.

**Comment by piotrtomiak** [comment]
@Tricky-Ricky do you get the issue when invoking from command line? If not, than it's most likely a problem in VSCode integration.

**Comment by Tricky-Ricky** [comment]
The issue is with the linter configuration in the Vue install.
I did a fresh install without the linter, wrote the same code with no errors, and it displayed fine.
Tried again with the linter installed on a fresh install, same error...

This is with @vue/cli 4.5.11

Choosing vue 2 with eslint + prettier config, including vuex and router.

The error shows in the terminal of VSCode.

The steps I took was to change the helloworld stuff, renamed the default file to hud, and relevant details, It would then show the error, at the bottom of the file, as per the image kawazoe posted above.

**Comment by kawazoe** [comment]
@Tricky-Ricky I'm not too sure if I understand your last post correctly.

> I did a fresh install without the linter, wrote the same code with no errors, and it displayed fine.
> Tried again with the linter installed on a fresh install, same error...

Since this error is produced by the linter, it is not surprising that not installing it will not produce the error. This is irrelevant of the extensions or integration features your IDE provides. If you do not configure eslint, WebStorm will not show an error either, because it uses eslint to show your those errors.

> The error shows in the terminal of VSCode.

Did you try running `npm run lint` in the terminal or did you try something else to get this error? Can you post the log either as text or as a screenshot?

**Comment by prady1900** [comment]
I have similar error tried running `npm run lint` but issue still persists. Just like @kawazoe mentioned at the bottom of the script tag. And I am not using webstorm either. Help please. IDE is VSCode.
![Screenshot (118)](https://user-images.githubusercontent.com/56378577/114299554-99921300-9ad9-11eb-9990-a897912b4984.png)

**Comment by Akryum** [comment]
I also get this error with `@rollup/plugin-eslint 8.0.1`

**Comment by StephanieTM** [comment]
Same error with VSCode and vue-cli, my solution here is to add 
```
public/index.html # or path to your html file with this problem
```
in my `.eslintignore` file.

**Comment by Mouvedia** [comment]
FYI when using `-c` option, the `"vue/comment-directive": 0` needs to be in the override.

**Comment by live680g** [comment]
> Same here.
> Only solution for me was to disable the rule
> 
> ```
>   rules: {
>     'vue/comment-directive': 'off'
>   }
> ```

its useful. thx

**Comment by dumptyd** [comment]
This is what I ended up doing

```js
// .eslintrc.js
module.exports = {
 // ...
  overrides: [
    {
      files: ['*.html'],
      rules: {
        'vue/comment-directive': 'off'
      }
    }
  ]
};
```

**Comment by Germxu** [comment]
I  added ***.html** to .eslintignore

**Comment by srsutherland** [comment]
A better solution may be to set up eslint to actually be able to lint `.html` files properly (which vue should probably do by default, but whatever). Try adding:

```js
{
  // (other stuff...),
  "plugins": [
    "html"
  ]
}
```

to your `.eslintrc.json` or `.eslintrc.cjs` file and run:

```
npm install --save-dev eslint-plugin-html
```

if you don't already have it.
